pkg/chunk/schema.go

```diff
@@ -536,3 +547,33 @@ func (v7Entries) GetReadQueries(bucket Bucket) ([]IndexQuery, error) {
 		},
 	}, nil
 }
+
+// v8Entries is the same as v7Entries however with a series index instead of a metric name index
+type v8Entries struct {
+	v6Entries
+}
+
+func (entries v8Entries) GetWriteEntries(bucket Bucket, metricName model.LabelValue, labels model.Metric, chunkID string) ([]IndexEntry, error) {
+	indexEntries, err := entries.v6Entries.GetWriteEntries(bucket, metricName, labels, chunkID)
+	if err != nil {
+		return nil, err
+	}
+
+	fingerprintBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(fingerprintBytes, uint64(labels.Fingerprint()))
+
+	seriesBytes, err := json.Marshal(labels)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I couldn't find a convenient function, no.

I don't like the idea of putting JSON in here, its a PITA to deal with backwards compatibility etc.  But I guess it'll have to do for now.